az-snp-vtpm/src/main.rs

```diff
@@ -52,10 +52,13 @@ fn main() -> Result<(), Box<dyn Error>> {
             let snp_report = hcl_report.snp_report();
 
             let (vcek, cert_chain) = if imds {
-                certs::get_from_imds()?
+                let certs = imds::get_certs()?;
+                let vcek = certs::Vcek(certs::X509::from_pem(&certs.vcek_cert)?);
+                let cert_chain = certs::build_chain(&certs.certificate_chain)?;
+                (vcek, cert_chain)
             } else {
-                let vcek = certs::get_vcek_from_amd(snp_report)?;
-                let cert_chain = certs::get_chain_from_amd()?;
+                let vcek = amd_kds::get_vcek(snp_report)?;
+                let cert_chain = amd_kds::get_chain()?;
```

Comment:
yes, I also wondered whether this could be aligned into a common trait (`CertProvider` or something), but the two methods are not really alike. the IMDS code will only work on a CVM, i.e. the attester side, is http only, and yields VCEK, ASK, ARK in a single request, while the kds has two discrete requests for cert chain and vcek and requires platform info from the report as input.

Reply:
Gotcha makes sense.